### DISCUSSION TITLE
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'icon_id' in 'field list'

### DISCUSSION THREAD

**Discussion by cchabot-besse** [dbarzin/mercator#893] [General]
Bonjour, suite à la dernière mise à jour de l'image Docker on a l'erreur `[SQLSTATE[42S22]: Column not found: 1054 Unknown column 'icon_id' in 'field list'` sur une tentative de modification ou d'exploration d'une entité

**Comment by dbarzin** [comment]
N'aurais-tu pas oublié de migrer la base de données ?

**Reply by cchabot-besse** [reply]
J'ai lancé la migration mais j'ai l'erreur suivante
```bash
ee263db139f3:/var/www/mercator$ php artisan migrate
   INFO  Running migrations.

  2024_05_15_212326_routers_log_phys ................................................................................................... 3.58ms FAIL

   Illuminate\Database\QueryException

  SQLSTATE[42S21]: Column already exists: 1060 Duplicate column name 'type' (Connection: mysql, SQL: alter table `routers` add `type` varchar(255) null)

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:825
    821▕                     $this->getName(), $query, $this->prepareBindings($bindings), $e
    822▕                 );
    823▕             }
    824▕
  ➜ 825▕             throw new QueryException(
    826▕                 $this->getName(), $query, $this->prepareBindings($bindings), $e
    827▕             );
    828▕         }
    829▕     }

      +9 vendor frames

  10  database/migrations/2024_05_15_212326_routers_log_phys.php:14
      Illuminate\Support\Facades\Facade::__callStatic("table")
      +25 vendor frames

  36  artisan:35
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
```

**Comment by dbarzin** [comment]
Cela signifie que cette migration a probablement déjà été appliquée sur cette base de données, mais que cela ne se trouve pas dans la table DB "migrations" qui contient la liste des migrations effectuées. Vous avez sans doute restauré un backup DB sur une autre version de l'application.

Le plus rapide sera de mettre en commentaire dans les fichiers de migration les modifications de table qui ont déjà été réalisées, et cela, jusqu'à ce que toutes les migrations soient passées.

**Reply by cchabot-besse** [reply]
D'accord merci.
Je ne suis pas grand connaisseur de Laravel, où se trouvent ces fichiers de migration des modifications de table ?

**Comment by dbarzin** [comment]
Dans le répertoire `database/migrations`

**Reply by cchabot-besse** [reply]
Merci pour le support, c'est corrigé
